File: 01elevator_python/gpio.py
##########################################################
# * Python GPIO Functions for Traffic Signal Simulation
# * using Baglebone Black running Debian 7 Linux distribution
##########################################################

import sys
import os


SYSFS_GPIO_DIR = "/sys/class/gpio"


def gpioUnexport (gpio):
	try: 
   		fo = open(SYSFS_GPIO_DIR + "/unexport","w")  
   		fo.write(gpio)
   		fo.close()
   		return
   	except IOError:
 		return
             	

def gpioExport (gpio): 
	try:
   		fo = open(SYSFS_GPIO_DIR + "/export","w")  
   		fo.write(gpio)
   		fo.close()
   		return
   	except IOError:
                return


def gpioSetDir (gpio, flag):
	try: 
	   	fo = open(SYSFS_GPIO_DIR + "/gpio" + gpio + "/direction" ,"w")  
	   	fo.write(flag)
	   	fo.close()
	   	return
 	except IOError:
                return


def gpioSetVal (gpio, val):
	try: 
		fo = open(SYSFS_GPIO_DIR + "/gpio" + gpio + "/value" ,"w")  
		fo.write(val)
		fo.close()
		return
	except IOError:
                return


def gpioSetEdge (gpio, flag): 
	try:
		fo = open(SYSFS_GPIO_DIR + "/gpio" + gpio + "/edge" ,"w")  
		fo.write(flag)
		fo.close()
   		return
	except IOError:
                return

File: 01elevator_cpp/gpio.cpp
/*******************************************************
 * C++ code for GPIO Functions for Lift Simulation using
 * Baglebone Black running Debian 7 Linux distribution
 ********************************************************/

#include<iostream>
#include<fstream>
#include<string>

using namespace std;


#define SYSFS_GPIO_DIR "/sys/class/gpio"


void gpioExport(string gpio)
{
	   fstream fs;
	   string path(SYSFS_GPIO_DIR);
	   fs.open((path + "/export").c_str(), fstream::out);
	   fs << gpio;
	   fs.close();
}


void gpioUnexport(string gpio)
{
	   fstream fs;
	   string path(SYSFS_GPIO_DIR);
	   fs.open((path + "/unexport").c_str(), fstream::out);
	   fs << gpio;
	   fs.close();
}

void gpioSetDir(string gpio, string dir)
{
	   fstream fs;
	   string path(SYSFS_GPIO_DIR);
	   fs.open((path + "/gpio" + gpio + "/direction").c_str(), fstream::out);
	   fs << dir;
	   fs.close();
}


void gpioSetEdge(string gpio, string edge)
{
	   fstream fs;
	   string path(SYSFS_GPIO_DIR);
	   fs.open((path + "/gpio" + gpio + "/edge").c_str(), fstream::out);
	   fs << edge;
	   fs.close();
}

void gpioSetValue(string gpio, string val)
{
	   fstream fs;
	   string path(SYSFS_GPIO_DIR);
	   fs.open((path + "/gpio" + gpio + "/value").c_str(), fstream::out);
	   fs << val;
	   fs.close();
}

